vchan/io.c

```diff
@@ -98,12 +99,11 @@ int libvchan_wait(libvchan_t *ctrl) {
          * right away).
          */
         while (ret == -2 && libxenvchan_is_open(ctrl->xenvchan) == 2) {
-            fd_set rd_set;
-            struct timeval tv = { 10, 0 };
             int vchan_fd = libxenvchan_fd_for_select(ctrl->xenvchan);
-            FD_ZERO(&rd_set);
-            FD_SET(vchan_fd, &rd_set);
-            switch (select(vchan_fd+1, &rd_set, NULL, NULL, &tv)) {
+            struct pollfd fds[] = {
+                { .fd = vchan_fd, .events = POLLIN | POLLOUT | POLLPRI | POLLRDHUP, .revents = 0 },
```

Comment:
`PULLRDHUP` is not applicable here. `PULLPRI` looks irrelevant for this FD too. `PULLOUT` is actively wrong here.
So, please keep only `PULLIN`.